pkg/util/names/names.go

```diff
@@ -0,0 +1,32 @@
+package names
+
+import (
+	"regexp"
+	"strings"
+
```

Comment:
Preferably we can create this `name` package outside `util`. At some point we should remove packages from inside `util` -- the way we currently use is an anti-pattern.

Reply:
It might make sense to just have this in [mongodbcommunity_types.go](https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/api/v1/mongodbcommunity_types.go) in this case.

Reply:
No, it would still be a utility function, but its only purpose right now is to normalise this particular secret name and it is called when generating `scram.Users` from a `MongoDBCommunity` resource.

Reply:
got it, makes sense to move there then.